iceoryx_posh/test/moduletests/test_popo_toml_gateway_config_parser.cpp

```diff
@@ -20,201 +20,532 @@
-    auto result = iox::config::StubbedTomlGatewayConfigParser::validate(*toml);
-    EXPECT_EQ(true, result.has_error());
-    if (result.has_error())
-    {
-        EXPECT_EQ(iox::config::TomlGatewayConfigParseError::INCOMPLETE_SERVICE_DESCRIPTION, result.get_error());
-    }
+    auto result = StubbedTomlGatewayConfigParser::validate(*toml);
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(TomlGatewayConfigParseError::INCOMPLETE_SERVICE_DESCRIPTION, result.get_error());
 }
 
-TEST_F(TomlGatewayConfigParserTest, FailsValidationIfServiceDescriptionBeginsWithNumber)
+TEST_F(TomlGatewayConfigParserTest, ServiceDescriptionBeginsWithNumberReturnInvalidServiceDescriptionError)
 {
-    // ===== Setup
-    // Prepare configuration to test with
     auto toml = cpptoml::make_table();
     auto serviceArray = cpptoml::make_table_array();
 
     auto serviceBeginsWithNumber = cpptoml::make_table();
     serviceBeginsWithNumber->insert("service", "0000");
     serviceBeginsWithNumber->insert("instance", "0000");
     serviceBeginsWithNumber->insert("event", "0000");
-    serviceBeginsWithNumber->insert("size", 0);
     serviceArray->push_back(serviceBeginsWithNumber);
     toml->insert("services", serviceArray);
 
-    // ===== Test
+    auto result = StubbedTomlGatewayConfigParser::validate(*toml);
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(TomlGatewayConfigParseError::INVALID_SERVICE_DESCRIPTION, result.get_error());
+}
+
+TEST_F(TomlGatewayConfigParserTest, HyphenInServiceDescriptionReturnInvalidServiceDescriptionError)
+{
+    auto toml = cpptoml::make_table();
+    auto serviceArray = cpptoml::make_table_array();
+
+    auto serviceEntryWithHyphen = cpptoml::make_table();
+    serviceEntryWithHyphen->insert("service", "service-name");
+    serviceEntryWithHyphen->insert("instance", "instance-name");
+    serviceEntryWithHyphen->insert("event", "event-name");
+    serviceArray->push_back(serviceEntryWithHyphen);
+    toml->insert("services", serviceArray);
+
+    auto result = StubbedTomlGatewayConfigParser::validate(*toml);
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(TomlGatewayConfigParseError::INVALID_SERVICE_DESCRIPTION, result.get_error());
+}
+
+TEST_F(TomlGatewayConfigParserTest, NoServicesInConfigReturnIncompleteConfigurationError)
+{
+    auto toml = cpptoml::make_table();
+
     auto result = iox::config::StubbedTomlGatewayConfigParser::validate(*toml);
-    EXPECT_EQ(true, result.has_error());
-    if (result.has_error())
-    {
-        EXPECT_EQ(iox::config::TomlGatewayConfigParseError::INVALID_SERVICE_DESCRIPTION, result.get_error());
-    }
+
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(TomlGatewayConfigParseError::INCOMPLETE_CONFIGURATION, result.get_error());
+}
+
+TEST_F(TomlGatewayConfigParserTest, ParseWithoutParameterTakeDefaultPathReturnNoError)
+{
+    auto result = TomlGatewayConfigParser::parse();
+    GatewayConfig config = result.value();
+
+    EXPECT_FALSE(result.has_error());
```

Comment:
You're accessing value without checking for error first/if a value is present.
You should move the `EXPECT` before using result/value and change it to `ASSERT_*.`
This is the case for all following tests below as well!